src/cargo/ops/cargo_update.rs

```diff
@@ -207,6 +221,215 @@ pub fn print_lockfile_changes(
+            |dependency| -> CargoResult<_> {
+                let name = dependency.package_name();
+                let renamed_to = dependency.name_in_toml();
+
+                if name != renamed_to {
+                    trace!(
+                        "skipping dependency renamed from `{}` to `{}`",
+                        name,
+                        renamed_to
+                    );
+                    return Ok(dependency);
+                }
+
+                let version_req = dependency.version_req();
+
+                if let OptVersionReq::Req(current) = version_req {
+                    if let [comparator] = &current.comparators[..] {
+                        if comparator.op == Op::Caret {
+                            let query = crate::core::dependency::Dependency::parse(
+                                name,
+                                None,
+                                dependency.source_id().clone(),
+                            )?;
+
+                            let possibilities = {
+                                loop {
+                                    match registry.query_vec(&query, QueryKind::Exact) {
+                                        std::task::Poll::Ready(res) => {
+                                            break res?;
+                                        }
+                                        std::task::Poll::Pending => registry.block_until_ready()?,
+                                    }
+                                }
+                            };
+
+                            let latest = if !possibilities.is_empty() {
+                                possibilities
+                                    .iter()
+                                    .map(|s| s.as_summary())
+                                    .map(|s| s.version())
+                                    .max()
+                            } else {
+                                None
+                            };
+
+                            if let Some(latest) = latest {
+                                if !current.matches(&latest) {
+                                    trace!(
+                                        "upgrading `{}` from {} to {}",
+                                        name,
+                                        current,
+                                        latest
+                                    );
+
+                                    opts.gctx.shell().status_with_color(
+                                        "Upgrading",
+                                        format!(
+                                            "{} {} -> v{}",
+                                            name,
+                                            current,
+                                            latest.to_string()
+                                        ),
+                                        &style::GOOD,
+                                    )?;
```

Comment:
We're reporting `Upgrading {name} {current_req} -> v{latest_version}`.  Should we be doing `Upgrading {name} {current_req} -> {new_req}`?

Reply:
Fixed.